xc/commerce/UpdateHostname.ps1

```diff
@@ -0,0 +1,170 @@
+        }
+    }
+
+    $json = ConvertTo-Json $json -Depth 100
+    Set-Content $configPath -Value $json -Encoding UTF8
+    Write-Host "Done patching $configPath!" -ForegroundColor Green
+}
+
+Function UpdateCommerceConfig() {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$configPath, 
+        [Parameter(Mandatory = $true)]
+        [string[]]$allowedOrigins,
+        [Parameter(Mandatory = $true)]
+        [string]$identityServerUrl
+    )
+
+    Write-Host "Patching $configPath"
+    $json = Get-Content $configPath -raw | ConvertFrom-Json;
+    Write-Host "Patching AllowedOrigin with $allowedOrigins"
+    $json.AppSettings.AllowedOrigins = $allowedOrigins
+    Write-Host "Patching SitecoreIdentityServerUrl with $identityServerUrl"
+    $json.AppSettings.SitecoreIdentityServerUrl = $identityServerUrl
+    $json = ConvertTo-Json $json -Depth 100
+    Set-Content $configPath -Value $json -Encoding UTF8
+    Write-Host "Done patching $configPath!" -ForegroundColor Green
+}
+
+#Update the hostfile with the new hostnames
+Write-Host "Updating hostfile with hostname $commerceHostname"
+$hostFileName = 'c:\\windows\\system32\\drivers\\etc\\hosts'; '\"`r`n127.0.0.1`t$commerceHostname\"' | Add-Content $hostFileName
+Write-Host "Succesfully updated hostfile!" -ForegroundColor Green
+
+#Modify the certificate with the new hostnames
+[X509Certificate[]]$certificates = Get-ChildItem -Path 'cert:\localmachine\my' -DnsName 'DO_NOT_TRUST_SitecoreRootCert';
+[X509Certificate]$rootCert = $certificates[0];
+Write-Host "Creating a new certificate with hostnames $commerceHostname"
+[X509Certificate]$certificate = New-SelfSignedCertificate -certstorelocation cert:\localmachine\my -dnsname $commerceHostname -Signer $rootcert -KeyExportPolicy Exportable -Provider 'Microsoft Enhanced RSA and AES Cryptographic Provider';
+Write-Host "Succesfully created the certificate!. Updating bindings..." -ForegroundColor Green
+
+UpdateBindingCertificate -certificate $certificate -bindingName 'CommerceOps_Sc9' -port 5015
+UpdateBindingCertificate -certificate $certificate -bindingName 'CommerceShops_Sc9' -port 5005
+UpdateBindingCertificate -certificate $certificate -bindingName 'CommerceAuthoring_Sc9' -port 5000 
+UpdateBindingCertificate -certificate $certificate -bindingName 'CommerceMinions_Sc9'-port 5010
+UpdateBindingCertificate -certificate $certificate -bindingName 'SitecoreBizFx' -port 4200
+UpdateBindingCertificate -certificate $certificate -bindingName 'SitecoreIdentityServer' -port 5050
+
+#Prepare some variables to be passed as arguments 
+[string]$baseUrl = "https://$commerceHostname"
+[string]$identityServerUrl = $baseUrl + ":5050"
+[string]$engineServerUrl = $baseUrl + ":5000"
+[string]$bizFxUrl = $baseUrl + ":4200"
+[string]$plumberUrl = $baseUrl + ":4000"
+$redirectUrls = [string[]]@(
+    ($bizFxUrl),
+    ($bizFxUrl + "/?")
+)
+$allowedOrigins = [string[]]@(
+    "http://$sitecoreHostname",
```

Comment:
I think this should remain to be set to the default `http://sitecore` iso `http://sitecore.local` as we do not change the Sitecore domain name.

Reply:
Actually sitecore can now be accessed at sitecore.local in our images.

https://github.com/avivasolutionsnl/sitecore-docker/pull/99/files#diff-d591119415fa54e2ca1f21deca0591eaR83

Reply:
No need to update the Sitecore sites config for that?